standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -2312,14 +2314,19 @@ private Table convertToTable(MTable mtbl) throws MetaException {
       } else {
         tableType = TableType.MANAGED_TABLE.toString();
       }
+    } else {
+      if (!tableType.equals(TableType.MANAGED_TABLE.toString()) && !tableType.equals(TableType.EXTERNAL_TABLE.toString())) {
```

Comment:
why not simply `TableType.VIRTUAL_VIEW.toString().equals(tableType)` ?

Reply:
There is a UT here that uses the table type `VIEW`, which causes the test to fail.
So I'm thinking if this table type might allow other non-enumeration values ​​to be stored.

https://github.com/apache/hive/blob/3ca97725854df10e7cd67514a3219e996453eab1/standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestTablesCreateDropAlterTruncate.java#L1598

Reply:
TableType is an enum that defines list of allowed values. Optimization that you've added is only relevant to the views. There could be some issue with test and we should fix it.